pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -220,7 +220,7 @@ func (r *ScaleREST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object
 	}
 
 	if deploymentStatus := util.DeploymentStatusFor(controller); deploymentStatus != api.DeploymentStatusComplete {
-		return nil, false, err
+		return nil, false, errors.NewConflict("Scale", scale.Name, fmt.Errorf("latest deployment \"%s\" is not complete or has failed: %v", controller.Name, deploymentStatus))
```

Comment:
I don't know that `NewConflict` is the best error to return.  Lots of places trigger off of conflicts to do GET, diff, merge, update flows.  Maybe `NewInternalError`?

@ironcladlou